apps/angular/pipe-easy/src/app/app.component.ts

```diff
@@ -1,21 +1,23 @@
 import { NgFor } from '@angular/common';
 import { Component } from '@angular/core';
+import { HeavyComputationPipe } from '../helpers/heaveyComputation.pipe';
 
 @Component({
   standalone: true,
-  imports: [NgFor],
+  imports: [NgFor, HeavyComputationPipe],
   selector: 'app-root',
   template: `
     <div *ngFor="let person of persons; let index = index">
-      {{ heavyComputation(person, index) }}
+      <!-- {{ heavyComputation(person, index) }} -->
+      {{ person | heavyComputation : index }}
     </div>
   `,
 })
 export class AppComponent {
-  persons = ['toto', 'jack'];
+  persons = ['toto', 'jack', 'jill', 'john', 'jane'];
 
-  heavyComputation(name: string, index: number) {
-    // very heavy computation
-    return `${name} - ${index}`;
-  }
+  // heavyComputation(name: string, index: number) {
+  //   // very heavy computation
+  //   return `${name} - ${index}`;
+  // }
```

Comment:
delete this. 🙏

Reply:
Deleted it. Thanks for reviewing. Sure it will be helpful if you write a guide on angular challenges.
I have a question, in which directory we should put our custom pipes and directives?

Reply:
What do you mean ? inside your project ? Really depends ?  Are you using Nx, angular CLI ?  There is no good or bad answer. But most of the time, if the pipe is used only once, it should be close to where it is used, otherwise inside a share folder